src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -298,12 +298,16 @@ def _resolve_sampler(self, dataloader: DataLoader, shuffle: bool, mode: Optional
 
             # update docs too once this is resolved
             trainer_fn = self.trainer.state.fn
-            if isinstance(sampler, DistributedSampler) and trainer_fn in (TrainerFn.VALIDATING, TrainerFn.TESTING):
+            if (
+                self.trainer.distributed_sampler_kwargs["num_replicas"] > 1
```

Comment:
You need to get this info from the sampler directly. The user could supply one manually.

Reply:
```suggestion
                sampler.num_replicas > 1
```
But needs to move after the instance check.

Reply:
agree! but wouldn't that create a conflict if they won't match?

Reply:
done!

Reply:
Up to the user. We have examples in the docs that show how to use a custom DistributedSampler.